packages/core/src/Stores/client-store.js

```diff
@@ -1523,18 +1523,33 @@ export default class ClientStore extends BaseStore {
     setDeviceData() {
```

Comment:
Could you also split out the logic for this to a helper function that is imported to client-store? the setDeviceData method is doing too many things at the moment

Perhaps something like:
```javascript
import { createDeviceDataObject }

setDeviceData() {
   const new_device_data = createDeviceDataObject(...);
   this.device_data = { ...this.device_data, ...new_device_data };
}
```